rudder-core/src/main/scala/com/normation/rudder/reports/ComplianceMode.scala

```diff
@@ -44,23 +44,66 @@ import net.liftweb.json._
  * - error_only: only report for repaired and error reports.
  */
 
-sealed trait ComplianceMode {
-  def name: String
+sealed trait ComplianceModeName {
+  val name : String
 }
 
-final case object FullCompliance extends ComplianceMode {
+final case object FullCompliance extends ComplianceModeName {
   val name = "full-compliance"
 }
-final case class ChangesOnly (
-  heartbeatPeriod : Int
-)extends ComplianceMode {
-  val name = ChangesOnly.name
-}
 
-object ChangesOnly {
+final case object ChangesOnly extends ComplianceModeName {
```

Comment:
no need for final on object, they are already (you can look at the bytecode for that, or it's certainly specified somewhere in some scala spec). But you can let it here, I does no arm.